### PR TITLE
Popover: add `box-sizing` reset style

### LINKED ISSUE #56190: Link Control UI: scrollbars showing on smaller screens
On smaller screens a horizontal scrollbar appears when you create a new draft for a page via the navigation link control UI.

Screen Capture on 2023-11-16 at 10-22-05

### LINKED ISSUE #58865: Link Control: drafting a page in nav block causes resizing issues
Using 6.5-alpha-57555, I noticed that if I created a drafted page in the navigation block, the link UI suddenly seems to struggle to resize and adds a scrollbar:

https://github.com/WordPress/gutenberg/assets/26996883/431e7a1b-0ec8-4d79-ba3d-dcdf93b2ad88

I'm using WP Playground in case that matters! cc @richtabor

### PR DESCRIPTION
**Pull request by t-hamano**
Fixes #58865
Fixes #56190
Alternative to #56325

## What?

This PR applies `box-sizing` reset style to the `Popover` component itself to prevent unintended horizontal scrollbars from appearing in any context. This will prevent unintended scroll bars from appearing in the popover of the navigation block in the Site Editor.

| Before | After |
|--------|--------|
| image| image | 

## Why?

Box sizing reset style for Popover was being applied on each editor instance, not on the component itself. However, unlike the post editor, this reset was not applied in the Site Editor.

https://github.com/WordPress/gutenberg/blob/9ca92adec8303df0094b6eebaeed714df69a356a/packages/edit-site/src/style.scss#L70-L74

I suspect that this is the cause of this issue.

## How?

Applied a box sizing reset style to the component itself and removed the reset mixin when it was no longer needed.

As a further step, we might also apply a similar approach to the `Modal` component.

https://github.com/WordPress/gutenberg/blob/31d5bd0167fb6667be6695a83bf6a94bc1701f3c/packages/edit-post/src/style.scss#L51-L53
https://github.com/WordPress/gutenberg/blob/31d5bd0167fb6667be6695a83bf6a94bc1701f3c/packages/edit-site/src/style.scss#L72-L74
https://github.com/WordPress/gutenberg/blob/31d5bd0167fb6667be6695a83bf6a94bc1701f3c/packages/edit-widgets/src/style.scss#L21-L22
## Testing Instructions

- Open the Site Editor and select the navigation block.
- Add a new link.
- The popover showing links should not display scrollbars.

### CODE REVIEW

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -26,6 +26,7 @@
 -   `Placeholder`: Fix Global Styles typography settings bleeding into placeholder component ([#58303](https://github.com/WordPress/gutenberg/pull/58303)).
 -   `PaletteEdit`: Fix palette item accessibility in details view ([#58214](https://github.com/WordPress/gutenberg/pull/58214)).
 -   `Snackbar`: Fix the auto-dismissal timers ([#58604](https://github.com/WordPress/gutenberg/pull/58604)).
+-   `Popover`, Add `box-sizing` reset style ([#58871](https://github.com/WordPress/gutenberg/pull/58871)).
```
Comment (andrewhayward) [inline packages/components/CHANGELOG.md]:
```suggestion
-   `Popover`: Add `box-sizing` reset style ([#58871](https://github.com/WordPress/gutenberg/pull/58871)).
```

**Review by andrewhayward — APPROVED** [review]
Works for me 🚀

I was pretty confident `reset` wasn't a valid `box-sizing` value, and initially very confused by the `box-sizing:reset` title! Might be worth adjusting that before merging to prevent any future mystery.

**Comment by t-hamano** [comment]
> I was pretty confident `reset` wasn't a valid `box-sizing` value, and initially very confused by the `box-sizing:reset` title!

Sorry for confusing you 😅 Is the current title appropriate?

**Comment by andrewhayward** [comment]
> Sorry for confusing you 😅 Is the current title appropriate?

Thanks! Much clearer now 👍

**Comment by youknowriad** [comment]
While I like this PR, It makes me think that popover is not the only component that should have this built-in. Almost all components should have it. What if you use any component outside edit-post or edit-site... I kind of wish we can remove the "reset" from edit-post, edit-site and popovers to get some clarity on which components actually need this.

**Comment by mirka** [comment]
> popover is not the only component that should have this built-in. Almost all components should have it.

Previous discussion of this issue here: #42415

**Pull request merged**